docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1737,6 +1737,200 @@ As a response to blocked request AdGuard returns a short video placeholder.
 >
 > Rules with `$mp4` modifier are not supported by AdGuard Content Blocker, AdGuard for iOS and Safari.
 
+### Rule priorities {#rule-priorities}
+
+Each rule has its own priority, which is necessary when several rules match the request and the filtering system needs to select one of them.
+Priority is measured by a positive integer.
+In the case of a conflict between two rules with the same priority value, it is unspecified which one of them will be chosen.
+
+#### Priority computation
+
+To calculate priority, modifiers have been divided into several groups. They are arranged in order of priority from lowest to highest. The more a modifier narrows the scope of a rule, the more priority it adds to the total weight of the rule or the more a rule covers requests, the lower its priority. Note that there are modifiers where setting one parameter has a higher priority than setting two or more, for example, for `$domain=example.com|example.org`, because a rule with two domains has a smaller, infinitesimal, effective area, but still larger than a rule with one resolved domain.
+
+The base priority weight of any rule is 1. If the calculated priority is a floating-point number, it will be **rounded up** to the smallest integer greater than or equal to the calculated weight.
+
+> **Compatibility with different versions of AdGuard**
+>
+> The calculation of numerical priorities for rules is effective in `tsurlfilter` since `v2.1.0`.
+
+#### 1. Basic modifiers, the presence of each adds 1 to the priority: {#priority-category-1}
+
+[//]: # (Please keep them sorted)
+
+ * [`$app`](#app-modifier) with forbidden applications using `~`,
+ * [`$denyallow`](#denyallow-modifier),
+ * [`$dnsrewrite`](#dnsrewrite-modifier),
+ * [`$domain`](#domain-modifier) with forbidden domains using `~`,
+ * `$first-party`,
+ * [`$header`](#header-modifier),
+ * [`$match-case`](#match-case-modifier),
+ * [`$method`](#method-modifier) with forbidden methods using `~`,
+ * [`$third-party`](#third-party-modifier),
+ * [`$to`](#to-modifier),
+ * restricted [content-types](#content-type-modifiers) with `~`.
+
+For forbidden domain, app or content-type, we add 1 for the presence of the modifier itself disregarding the number of forbidden domains or content-types, because the scope of the rule is infinitely large anyway. In other words, by banning several domains and content-types, we only *insignificantly* narrowed the scope of the rule.
```

Comment:
method and header as well

Reply:
When writing documentation avoid talking in first person.

`we narrowed the scope of the rule` --> `the scope of the rule is narrowed`

Reply:
@ameshkov we can't use negative values in `$header`, can we?

Reply:
rewritten this block with chatgpt